Rework the docs for the new world of Chef specific rules (#122)
chef/cookstyle: README.md

* **Issues [Response Time Maximum](https://github.com/chef/chef-oss-practices/blob/master/repo-management/repo-states.md):** 14 days
* **Pull Request [Response Time Maximum](https://github.com/chef/chef-oss-practices/blob/master/repo-management/repo-states.md):** 14 days

Cookstyle is a [code linting](https://en.wikipedia.org/wiki/Lint_%28software%29) tool that helps you to write better Chef Infra cookbooks by detecting and automatically correcting style, syntax, and logic mistakes in your code.

Cookstyle is powered by the [RuboCop](http://www.rubocop.org) linting engine. RuboCop ships with over three-hundred rules, or cops, designed to detect common Ruby coding mistakes and enforce a common coding style. We've customized Cookstyle with a subset of those cops that we believe are perfectly tailored for cookbook development. We also ship Chef specific cops that catch common cookbook coding mistakes, cleanup portions of code that are no longer necessary, and detect deprecations that prevent cookbooks from running on the latest releases of Chef Infra Client.

## Cookstyle vs. RuboCop

How does Cookstyle differ from RuboCop?

#### Tailored for Cookbooks

> Cookbook development differs from that of traditional Ruby software development so we have tailored the list of built in cops in RuboCop for cookbook development. For the most part, this means disabling cops we don't think are useful for cookbook development. Occcasionally we've changed the configuration of a rule to enforce a different behavior altogether. We've also extended the base RuboCop package with a set of our own Chef Infra specific cops. These cops are only found in Cookstyle and will help you to write more reliable and future-proof cookbooks.

Reviewer: ```suggestion
Cookbook development differs from that of traditional Ruby software development, so we have tailored the list of built-in cops in RuboCop for cookbook development. For the most part, this means disabling cops deemed not useful for cookbook development. Occasionally, we've changed the configuration of a rule to enforce a different behavior altogether. We've also extended the base RuboCop package with a set of our own Chef Infra-specific cops. These cops are only found in Cookstyle and will help you to write more reliable and future-proof cookbooks.
```

Author: Fixed

See the current set of cops in [config/cookstyle.yml](https://github.com/chef/cookstyle/blob/master/config/cookstyle.yml).

#### Stable

> RuboCop is an incredibly active project with new cops being introduced monthly. The new cops cause existing codebases to fail CI tests, and force authors to constantly update their code. With Cookstyle we update the RuboCop engine for bug and performance fixes, but we only change the set of cops tat will fail tests once a year during Chef Infra's April major release. All new cops are introduced at RuboCop's "refactor" alert level, meaning they will alert to the screen as you run Cookstyle, but they won't fail a build. This stability means you are free to upgrade releases of Cookstyle without being forced to update your infrastructure code.

Reviewer 2: `tat will fail` -> `that will fail`

Author: Fixed

> RuboCop is an incredibly active project with new cops being introduced monthly. The new cops cause existing codebases to fail CI tests and force authors to constantly update their code. With Cookstyle we update the RuboCop engine for bug and performance fixes, but we only change the set of cops that will fail tests once a year during Chef Infra's April major release. All new cops are introduced at RuboCop's "refactor" alert level, meaning they will alert to the screen as you run Cookstyle, but they won't fail a build. This stability means you are free to upgrade releases of Cookstyle without being forced to update your infrastructure code.

Reviewer: ```suggestion
RuboCop is an incredibly active project with new cops being introduced monthly. The new cops cause existing codebases to fail CI tests and force authors to constantly update their code. With Cookstyle, we update the RuboCop engine for bug and performance fixes, but we only change the set of cops that will fail tests once a year during Chef Infra's April major release. All new cops are introduced at RuboCop's "refactor" alert level, meaning they will alert to the screen as you run Cookstyle, but they won't fail a build. This stability means you are free to upgrade releases of Cookstyle without being forced to update your infrastructure code.
```

end
```

### From RuboCop

> Run RuboCop as normal, simply add a `-r cookstyle` option when running:

Reviewer: ```suggestion
Run RuboCop as normal, and simply add a `-r cookstyle` option when running:
```

### .rubocop.yml

As with RuboCop, any custom settings can still be placed in a `.rubocop.yml` file in the root of your project.

## Getting Involved

> We'd love to have your help developing Cookstyle. See our [Contributing Guide](https://github.com/chef/chef/blob/master/CONTRIBUTING.md) for more information on contributing to Chef projects. There's also a [Developer Guide](./DEVELOPER_GUIDE.md) for Cookstyle that outlines how the configs work and how you can upgrade the RuboCop engine.

Reviewer: ```suggestion
We'd love to have your help in developing Cookstyle. See our [Contributing Guide](https://github.com/chef/chef/blob/master/CONTRIBUTING.md) for more information on contributing to Chef projects. There's also a [Developer Guide](./DEVELOPER_GUIDE.md) for Cookstyle that outlines how the configs work and how you can upgrade the RuboCop engine.
```
